group-coordinator/src/test/java/org/apache/kafka/coordinator/group/ShareGroupAutoOffsetResetStrategyTest.java

```diff
@@ -0,0 +1,114 @@
+        assertThrows(ConfigException.class, () -> validator.ensureValid("test", "by_duration"));
+        assertThrows(ConfigException.class, () -> validator.ensureValid("test", "LATEST"));
+        assertThrows(ConfigException.class, () -> validator.ensureValid("test", "EARLIEST"));
+        assertThrows(ConfigException.class, () -> validator.ensureValid("test", "NONE"));
+        assertThrows(ConfigException.class, () -> validator.ensureValid("test", ""));
+        assertThrows(ConfigException.class, () -> validator.ensureValid("test", null));
+    }
+
+    @Test
+    public void testEqualsAndHashCode() {
+        ShareGroupAutoOffsetResetStrategy earliest1 = ShareGroupAutoOffsetResetStrategy.fromString("earliest");
+        ShareGroupAutoOffsetResetStrategy earliest2 = ShareGroupAutoOffsetResetStrategy.fromString("earliest");
+        ShareGroupAutoOffsetResetStrategy latest1 = ShareGroupAutoOffsetResetStrategy.fromString("latest");
+
+        ShareGroupAutoOffsetResetStrategy duration1 = ShareGroupAutoOffsetResetStrategy.fromString("by_duration:P2D");
+        ShareGroupAutoOffsetResetStrategy duration2 = ShareGroupAutoOffsetResetStrategy.fromString("by_duration:P2D");
+
+        assertEquals(earliest1, earliest2);
+        assertNotEquals(earliest1, latest1);
+        assertEquals(earliest1.hashCode(), earliest2.hashCode());
+        assertNotEquals(earliest1.hashCode(), latest1.hashCode());
+
+        assertNotEquals(latest1, duration2);
+        assertEquals(duration1, duration2);
+    }
+
+    @Test
+    public void testTimestamp() {
+        ShareGroupAutoOffsetResetStrategy earliest1 = ShareGroupAutoOffsetResetStrategy.fromString("earliest");
+        ShareGroupAutoOffsetResetStrategy earliest2 = ShareGroupAutoOffsetResetStrategy.fromString("earliest");
+        assertEquals(Optional.of(ListOffsetsRequest.EARLIEST_TIMESTAMP), earliest1.timestamp());
+        assertEquals(earliest1, earliest2);
+
+        ShareGroupAutoOffsetResetStrategy latest1 = ShareGroupAutoOffsetResetStrategy.fromString("latest");
+        ShareGroupAutoOffsetResetStrategy latest2 = ShareGroupAutoOffsetResetStrategy.fromString("latest");
+        assertEquals(Optional.of(ListOffsetsRequest.LATEST_TIMESTAMP), latest1.timestamp());
+        assertEquals(latest1, latest2);
+
+        ShareGroupAutoOffsetResetStrategy byDuration1 = ShareGroupAutoOffsetResetStrategy.fromString("by_duration:PT1H");
+        Optional<Long> timestamp = byDuration1.timestamp();
+        assertTrue(timestamp.isPresent());
+        assertTrue(timestamp.get() <= Instant.now().toEpochMilli() - Duration.ofHours(1).toMillis());
+
+        ShareGroupAutoOffsetResetStrategy byDuration2 = ShareGroupAutoOffsetResetStrategy.fromString("by_duration:PT1H");
+        ShareGroupAutoOffsetResetStrategy byDuration3 = ShareGroupAutoOffsetResetStrategy.fromString("by_duration:PT2H");
+
+        assertEquals(byDuration1, byDuration2);
+        assertNotEquals(byDuration1, byDuration3);
+    }
+}
```

Comment:
nit: Blank line at the end of the file please.

Reply:
ditto